src/stack_mitosis/interpreter.clj

```diff
@@ -18,6 +18,7 @@
 
 (defn databases
   [rds]
+  {:post [(not (empty? %))]}
```

Comment:
what does this line do? (commit is titled "add post condition for empty databases", but syntactically I'm not quite sure what this means)

Reply:
It's a post-condition for the function. :pre, and :post allow you to make a list of functions for asserting properties about the arguments and return values of a function. So in this case it asserts that the return value, represented by %, is not empty after fetching the list of databases from AWS. We were encountering that with the particular permissions we had for Jenkins, and it seemed to make sense to push the precondition earlier in the program to identify the source. I think we can do a better job of error handling and reporting invalid expectations, but this at least helps narrow the field for now.

https://clojure.org/reference/special_forms#_fn_name_param_condition_map_expr_2 is the official docs on pre/post conditions, but I think http://blog.fogus.me/2009/12/21/clojures-pre-and-post/ is a better explanation.

Reply:
ah interesting, thanks!